src/main/scala/com/databricks/spark/sql/perf/mllib/BenchmarkAlgorithm.scala

```diff
@@ -67,9 +69,10 @@ trait ScoringWithEvaluator {
   final override def score(
       ctx: MLBenchContext,
       testSet: DataFrame,
-      model: Transformer): Double = {
+      model: Transformer): MLMetrics = {
     val eval = model.transform(testSet)
-    evaluator(ctx).evaluate(eval)
+    // TODO: add MetricName for evaluators
```

Comment:
Cannot leave this TODO here. We should check if evaluator hasMetricName, if yes, use the metric name, otherwise the class name of the evaluator.